test/cpp/tensorexpr/test_loopnest.cpp

```diff
@@ -1117,5 +1117,33 @@ void testLoopNestComputeAt_4() {
   // TODO: Verify that computeAt works with reduction axis
 }
 
+void testOuterLoopVectorization() {
+  KernelScope kernel_scope;
+  Tensor* tensor = Compute(
+      "f", {{8, "X"}, {8, "y"}}, [](const VarHandle& x, const VarHandle& y) {
+        return ExprHandle(1.0f) + cast<float>(x) * x + cast<float>(y) * y;
+      });
+  LoopNest l({tensor});
+
+  Stmt* outer_loop = *dynamic_cast<Block*>(l.root_stmt())->stmts().begin();
+  l.vectorize(dynamic_cast<For*>(outer_loop));
+
+  Stmt* root_stmt = l.root_stmt();
+  Block* outer_block = dynamic_cast<Block*>(root_stmt);
+  ASSERT_NE(outer_block, nullptr);
+  while (Block* inner_block = dynamic_cast<Block*>(*outer_block->stmts().begin())) {
```

Comment:
How could we get a nested Blocks structure? If the intention was to go through `For` statements to the innermost body, then we need to look for alternating `For` and `Block` statements (rather than for repeating `Block` statements).

Reply:
We definitely get nested `Block` structures.  I had to write similar code in `kernel.cpp` as well.  Maybe we should consider moving `Block` out of the `Stmt` hierarchy to avoid this?

Reply:
Interesting, do you know how it occurs? 

> Maybe we should consider moving Block out of the Stmt hierarchy to avoid this?

That's an interesting idea! I think so far we've planned to handle this by inlining blocks into each other. It's not done yet, but @nickgg was going to work on that I believe.

Reply:
I don't know why it occurs, but like I said this isn't the first time I've encountered it.

Reply:
I'm working on a diff which adds Block flattening to the IR Simplifier, so we should be able to ignore this on code that happens after simplification.